src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs

```diff
@@ -631,8 +631,20 @@ impl CloudHypervisorInner {
     }
 
     pub(crate) fn stop_vm(&mut self) -> Result<()> {
+        // If the container workload exits, this method gets called. However,
+        // the container manager always makes a ShutdownContainer request,
+        // which results in this method being called potentially a second
+        // time. Without this check, we'll return an error representing EPIPE
+        // since the CH API socket is at that point invalid.
+        if self.state != VmmState::VmRunning {
```

Comment:
I was going to suggest to explicitly check for Stopped State here, but looks like there are only 3 states defined. I would eventually like to explicitly distinguish between NotReady and Stopped, but this would work for now.